### ISSUE TITLE
sp_Blitz - Using SkipChecks table for specific IDs excludes additional checks

### ISSUE THREAD

**Issue by MosesOfJ** [BrentOzarULTD/SQL-Server-First-Responder-Kit#877]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
I have defined a specific check to skip on a single database using a SkipChecks table.  When running sp_Blitz, additional checks are also also excluded.

I have so far discovered that the following checks are excluded when a specific ID has been defined:

* 14
* 21
* 26
* 75
* 77

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via http://sqlfiddle.com**

CREATE DATABASE [Maintenance];
GO

USE [Maintenance]
GO

CREATE TABLE dbo.BlitzChecksToSkip (ID INT IDENTITY(1,1) PRIMARY KEY, ServerName NVARCHAR(128), DatabaseName NVARCHAR(128), CheckID INT);

-- Exclude 'Last good DBCC CHECKDB over 2 weeks old' check for READONLY / OFFLINE databases
INSERT INTO dbo.BlitzChecksToSkip(DatabaseName, CheckID) SELECT [name], 68 FROM sys.databases WHERE is_read_only = 1 OR state_desc = 'OFFLINE';

EXEC dbo.sp_Blitz @SkipChecksDatabase='Maintenance', @SkipChecksSchema='dbo', @SkipChecksTable='BlitzChecksToSkip';

**What is the expected behavior?**
Only the configured checks are skipped on the specified database, all other checks on the database are included.

**Which versions of SQL Server and which OS are affected by this issue? Did this work in previous versions of our procedures?**
SQL Server 2012 on Windows Server 2012.

This was an issue at least back to VersionDate 20161022

**Comment by BrentOzar** [comment]
Forgive me, I'm feeling a little dumb here, and I need to try to reword this to understand it. I think I'm missing something obvious, so thanks in advance for your patience.

Instead of the select from sys.databases, can you do me a huge favor? Can you explicitly put a single row in that BlitzChecksToSkip table, with a specific name in there? Only do it for one of your read-only (or offline) databases, and then see what happens?

Reading the current code (v5.3,  20170501) I don't understand how this could happen, and I've done a quick repro on my system to verify that it's not happening, so there seems to be something unusual about your system. I want to make sure that it isn't involving something with the way you're populating the skip table.

Thanks!

**Comment by BrentOzar** [comment]
Ooo! As soon as I hit comment, of course I figure it out.

In the current sp_Blitz (v5.3, 20170501), check 77 has this line:

AND dSnap.name NOT IN (
SELECT DISTINCT
DatabaseName
FROM
#SkipChecks )

That's what's causing it to skip. Doh! Got it. Thanks!
